File: PROCESSING/NatureOfCode/0_Introduction/Gaussian/src/Sketch.java
import processing.core.PApplet;

import java.util.Random;

public class Sketch extends PApplet {

	public static void main(String[] args) {
		PApplet.main("Sketch");
	}
	
	
	Random random = new Random();
	

	@Override public void settings() {
		size(400, 400);
	}

	@Override public void setup() {
		
	}

	@Override public void draw() {
		float num = (float) random.nextGaussian();
		float standard_deviation = 60;
		float mean = 200;
		
		float x = standard_deviation * num + mean;
		
		noStroke();
		fill(0, 10);
		ellipse(x, 200, 16, 16);
	}
}

File: PROCESSING/NatureOfCode/9_GeneticAlgorithms/RandomString/src/Sketch.kt
import processing.core.PApplet
import java.util.*

// ultra retarded bruteforcer with random string

class Sketch : PApplet() {

	val rand = Random()
	val target = "cat"
	
	var active = true

	fun getRandomString(length : Int) : String {
		return String(CharArray(length) {
			'a' + rand.nextInt(26)
		})
	}

	override fun settings() {
		size(400, 400)
	}

	override fun setup() {
		fill(0)
		noStroke()
		frameRate(1000f)
	}

	override fun draw() {
		background(200)
		textSize(30f)
		
		val s = getRandomString(target.length)
		
		if (active) {
			if (s.equals(target)) active = false
			text("ACTIVE", 30f, 30f)
		}
		
		else {
			text("COMPLETE", 30f, 30f)
		}
		
		text(s, 30f, 100f)
		textSize(10f)
		text("also this is gonna never quess it so yor just wasting yor time", 30f, 150f)
	}
	
}

fun main(args: Array<String>) {
	PApplet.main("Sketch")
}